lib/charms/istio_k8s/v0/istio_ingress_config.py

```diff
@@ -252,6 +270,21 @@ def get_provider_ext_authz_info(
             log.debug("Failed to validate provider data: %s", e)
             return None
 
+    def is_fake_authz_config(self, relation: Relation) -> bool:
+        """Check if the provider has published a fake external authorization configuration.
+
+        Returns:
+            True if any of the provider relations contains fake authz configuration, else False.
+        """
+        provider_info = self.get_provider_ext_authz_info(relation)
+        if provider_info is not None:
+            if (
+                provider_info.ext_authz_service_name == FAKE_EXT_AUTHZ_SERVICE_NAME
+                and provider_info.ext_authz_port == FAKE_EXT_AUTHZ_PORT
+            ):
+                return True
+        return False
+
     def is_provider_ready(self, relation: Relation) -> bool:
         """Guard to check if the provider has published its external authorizer service configuration.
 
```

Comment:
sorry I should have caught this before.  This is a check to see if the data we are receiving on this relation is here?  Can't we just say `is_ready`?